bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -0,0 +1,265 @@
+        if (scanJob == null) {
+            scanJob = scheduler.schedule(this::scan, 1, TimeUnit.SECONDS);
+        }
+    }
+
+    void stopScan() {
+        Future<?> tmpScanJob = scanJob;
+        if (tmpScanJob != null) {
+            if (!tmpScanJob.isDone()) {
+                logger.trace("Trying to cancel IP scan");
+                tmpScanJob.cancel(true);
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException ignore) {
+                }
+            }
+            scanJob = null;
+        }
+    }
+
+    void scan() {
+        logger.warn("IpAddonFinder::scan");
+        for (AddonInfo candidate : addonCandidates) {
+            for (AddonDiscoveryMethod method : candidate.getDiscoveryMethods().stream()
+                    .filter(method -> SERVICE_TYPE.equals(method.getServiceType())).toList()) {
+
+                logger.trace("Checking candidate: {}", candidate.getUID());
+
+                Map<String, String> parameters = method.getParameters().stream()
+                        .collect(Collectors.toMap(property -> property.getName(), property -> property.getValue()));
+                Map<String, String> matchProperties = method.getMatchProperties().stream()
+                        .collect(Collectors.toMap(property -> property.getName(), property -> property.getRegex()));
+
+                // parse standard set op parameters:
+                String type = Objects.toString(parameters.get("type"), "");
+                String request = Objects.toString(parameters.get("request"), "");
+                String response = Objects.toString(matchProperties.get("response"), "");
+                int timeoutMs = 0;
+                try {
+                    timeoutMs = Integer.parseInt(Objects.toString(parameters.get("timeout_ms")));
+                } catch (NumberFormatException e) {
+                    logger.info("{}: discovery-parameter timeout_ms cannot be parsed", candidate.getUID());
+                    continue;
+                }
+                @Nullable
+                InetAddress destIp = null;
+                try {
+                    destIp = InetAddress.getByName(parameters.get("dest_ip"));
+                } catch (UnknownHostException e) {
+                    logger.info("{}: discovery-parameter dest_ip cannot be parsed", candidate.getUID());
+                    continue;
+                }
+                int destPort = 0;
+                try {
+                    destPort = Integer.parseInt(Objects.toString(parameters.get("dest_port")));
+                } catch (NumberFormatException e) {
+                    logger.info("{}: discovery-parameter dest_port cannot be parsed", candidate.getUID());
+                    continue;
+                }
+
+                //
+                // handle known types
+                //
+                try {
+                    switch (Objects.toString(type)) {
+                        case "ip_multicast":
```

Comment:
Define in a constant. Also should the type be in camel case?

Reply:
I looked at the metadata we use in addons.xml, but CamlCase is also fine if you like

Reply:
I would prefer camel case for consistency between finders. I called the parameter value for mDNS mdnsServiceType. Field names in xml are snake case, but content of fields is not defined.

Reply:
ok, now everything is camelCase (with lower case first letter)